src/org/joni/Search.java

```diff
@@ -421,10 +421,10 @@ final int search(Matcher matcher, byte[]text, int textP, int textEnd, int textRa
             int targetP = regex.exactP;
             int targetEnd = regex.exactEnd;
 
-            int end, s, tlen1;
+            int end, s;
             int tail = targetEnd - 1;
+            int tlen1 = tail - targetP;
             if (USE_SUNDAY_QUICK_SEARCH) {
-                tlen1 = tail - targetP;
```

Comment:
Is this fixing a separate bug? If so we should move it to another PR because it's not config-related.

Reply:
when I changed `USE_SUNDAY_QUICK_SEARCH ` to be initialized from system property it no longer compiled 

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.0:compile (default-compile) on project joni: Compilation failure: Compilation failure:
[ERROR] /Users/nezda/code/joni.git/src/org/joni/Search.java:[438,59] variable tlen1 might not have been initialized
[ERROR] /Users/nezda/code/joni.git/src/org/joni/Search.java:[446,59] variable tlen1 might not have been initialized
```

so I tried to fix.

Reply:
I pinged @enebo to look at this too.

Reply:
yeah, it was constant propagated since those interface fields are essentially static finals